python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -32,8 +34,14 @@ def cache(cache_file, data, base_configuration, hash_data, engine_name):
         contents_hash = hash_data[pc["id"]]["contents_hash"]
 
         toolkit_mgr.pipeline_configuration = pc["id"]
-        engine = toolkit_mgr.bootstrap_engine(engine_name, entity=entity)
         pc_descriptor = toolkit_mgr.resolve_descriptor(project_entity)
+        pc_path = pc_descriptor.get_path()
+
+        if glob.glob(os.path.join(pc_path, "config", "env", "shotgun_*.yml")):
```

Comment:
related to - https://github.com/shotgunsoftware/tk-core/pull/444 - would be great if we could get this logic to be on the inside of core instead so we can change it more easily. 

I imagine something along the lines of:

- bootstrap via start engine()
- (note: important - the toolkit manager does not know anything about the shotgun engine - it's completely agnostic!)
- start engine runs and pick environment resolves into an environment which does not contain a shotgun engine.
- in that case, we kick into legacy mode and look for shotgun_xxxx stuff (old pathways)

Would something like that be feasible you think?